lib/parsers/package-lock-parser.ts

```diff
@@ -49,66 +64,240 @@ export class PackageLockParser implements LockfileParser {
       version: manifestFile.version || '',
     };
 
-    const topLevelDeps: Dep[] = getTopLevelDeps(manifestFile, includeDev);
-
     // asked to process empty deps
     if (_.isEmpty(manifestFile.dependencies) && !includeDev) {
       return depTree;
     }
 
-    await Promise.all(topLevelDeps.map(async (dep) => {
-      depTree.dependencies[dep.name] = await this.buildSubTreeRecursiveFromPackageLock(
-        dep.name, ['dependencies'], packageLock, [], dep.dev);
-    }));
+    // prepare a flat map, where dependency path is a key to dependency object
```

Comment:
comment about the `if` above, unrelated to this PR: what will happen if both `dependencies` and `devDepenedencies` are empty?